python_modules/dagster-test/setup.py

```diff
@@ -1,26 +1,25 @@
 from setuptools import find_packages, setup
 
-if __name__ == "__main__":
-    setup(
-        name="dagster-test",
-        version="0+dev",
-        author="Elementl",
-        author_email="hello@elementl.com",
-        license="Apache-2.0",
-        description="A Dagster integration for test",
-        url="https://github.com/dagster-io/dagster/tree/master/python_modules/libraries/dagster-test",
-        classifiers=[
-            "Programming Language :: Python :: 3.7",
-            "Programming Language :: Python :: 3.8",
-            "Programming Language :: Python :: 3.9",
-            "Programming Language :: Python :: 3.10",
-            "License :: OSI Approved :: Apache Software License",
-            "Operating System :: OS Independent",
-        ],
-        packages=find_packages(exclude=["dagster_test_tests*"]),
-        install_requires=[
-            "dagster",
-            "pyspark",
-        ],
-        zip_safe=False,
-    )
+setup(
```

Comment:
how is this currently manifesting? should we log / fail somehow to prevent future similar mistakes?

Reply:
I think I'm going to add a check to make sure our setup.py files all conform to this (basically, just make sure there's no `__main__` string in any setup.py file).

Otherwise, when we go to parse the distribution, we get back an empty distribution so we don't know about the package's `install_requires` or `extras`.

Reply:
I spent a little bit of time trying to monkeypatch our distribution parsing to make it look like we're running as `__main__` but decided it wasn't worth it - there are some more notes in https://github.com/dagster-io/dagster/pull/9889/commits/4b6abcfe30ef46831d7a60e74fe3469a5ed427c1

Reply:
> when we go to parse the distribution, we get back an empty distribution

could we raise in this condition?

Reply:
Well, it's not really an _empty_ distribution. It's just whatever you get by default when you call `setup()`. I can look and see if there's any obvious tell that differentiates it from an otherwise vanilla package (like if we have a package with no dependencies).